sdk/core/Azure.Core/src/Extensions/IAzureClientFactoryBuilder.cs

```diff
@@ -8,6 +8,6 @@ namespace Azure.Core.Extensions
 {
     public interface IAzureClientFactoryBuilder
     {
-        IAzureClientBuilder<TClient, TOptions> RegisterClientFactory<TClient, TOptions>(Func<TOptions, TClient> clientFactory) where TOptions : ClientOptions;
+        IAzureClientBuilder<TClient, TOptions> RegisterClientFactory<TClient, TOptions>(Func<TOptions, TClient> clientFactory) where TOptions : class;
```

Comment:
Should we consider making the options always have a default constructor and using that constraint, or is that unimportant down the line?

Reply:
Unfortunately, the almost never have a default constructor. Instead, the guidelines prescribe having a constructor with ServiceVersion parameter defaulted to latest.

Reply:
Ah, yes.  Forgot about that due to the deviation in the Event Hubs service design.